openstack/networking/v2/subnets/requests.go

```diff
@@ -112,15 +121,24 @@ type UpdateOptsBuilder interface {
 // UpdateOpts represents the attributes used when updating an existing subnet.
 type UpdateOpts struct {
 	Name           string      `json:"name,omitempty"`
-	GatewayIP      string      `json:"gateway_ip,omitempty"`
+	GatewayIP      *string     `json:"gateway_ip,omitempty"`
```

Comment:
In this case, I think reverting `UpdateOpts` is the best choice. While it makes for being inconsistent with `CreateOpts`, it more naturally enforces allowed values. I think that's the style more favored here.

I'll update the code, but won't push anything until confirmation.